rs/nns/sns-wasm/src/sns_wasm.rs

```diff
@@ -1026,57 +1026,62 @@ where
                     vec![this_canister_id, canisters.root.unwrap()],
```

Comment:
I want to attach this comment to the triple slash comment above, but GH does not allow this...

There is an unstated assumption: we are the the exclusive controller of the canisters (before the method is called).

(If this does not hold, then the results will not be what the name says.)

Reply:
Clarified in a offline discussion: The name of this function is misleading, and understanding it requires reading the (only) place where it's called. Let's refactor this outside of this PR